packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -398,11 +425,31 @@ export function useConnections({
       trackConnectionAttemptEvent(connectionInfo);
       debug('connecting with connectionInfo', connectionInfo);
 
+      let notifyDeviceFlow:
+        | ((deviceFlowInformation: {
+            verificationUrl: string;
+            userCode: string;
+          }) => void)
+        | undefined;
+      if (isOIDCAuth(connectionInfo.connectionOptions.connectionString)) {
+        notifyDeviceFlow = (deviceFlowInformation: {
+          verificationUrl: string;
+          userCode: string;
+        }) => {
+          dispatch({
+            type: 'oidc-attempt-connect-notify-device-auth',
+            verificationUrl: deviceFlowInformation.verificationUrl,
+            userCode: deviceFlowInformation.userCode,
+          });
+        };
+      }
+
       const newConnectionDataService = await newConnectionAttempt.connect(
-        adjustConnectionOptionsBeforeConnect(
-          connectionInfo.connectionOptions,
-          appName
-        )
+        adjustConnectionOptionsBeforeConnect({
+          connectionOptions: connectionInfo.connectionOptions,
+          defaultAppName: appName,
+          notifyDeviceFlow,
```

Comment:
I think you’ll need to replace `getMongoClientConnectionOptions()` in DataServiceImpl with 

```
  return _.omit(this._mongoClientConnectionOptions, 'options.oidc.notifyDeviceFlow');
```

so that the shell doesn’t choke on the function property